promql/engine.go

```diff
@@ -310,25 +339,25 @@ func (ng *Engine) exec(ctx context.Context, q *query) (Value, error) {
 	ctx, cancel := context.WithTimeout(ctx, ng.timeout)
 	q.cancel = cancel
 
-	execTimer := q.stats.GetTimer(stats.ExecTotalTime).Start()
-	defer execTimer.Stop()
-	queueTimer := q.stats.GetTimer(stats.ExecQueueTime).Start()
+	execSpanTimer, ctx := newSpanTimer(ctx, "promqlExec", q.stats.GetTimer(stats.ExecTotalTime))
+	defer execSpanTimer.Finish()
+
+	queueSpanTimer, _ := newSpanTimer(ctx, "promqlExecQueue", q.stats.GetTimer(stats.ExecQueueTime), ng.metrics.queryQueueTime)
 
 	if err := ng.gate.Start(ctx); err != nil {
 		return nil, err
 	}
 	defer ng.gate.Done()
 
-	queueTimer.Stop()
-	ng.metrics.queryQueueTime.Observe(queueTimer.ElapsedTime().Seconds())
+	queueSpanTimer.Finish()
```

Comment:
Is it a problem that the if the `gate.Start` fails, this won't be stopped?

Reply:
Hmm, yep. But it wasn't stopped before either....